tests/test_catch.py

```diff
@@ -153,6 +153,21 @@ def handler(exc):
             raise ExceptionGroup("booboo", [ValueError("bar")])
 
 
+def test_catch_handler_reraises():
+    def handler(exc):
+        raise
```

Comment:
Can you add a test with `raise Exception` and make sure the `__context__` is set to the split exception group

Reply:
The `__context__` value is the same as `__cause__` on py3.11 though?

Reply:
So it seems that in the backport, `__context__` is set to the split exception group which doesn't match the py3.11 test which I just augmented.

Reply:
I made sure that the back-port works the same way as py3.11 does.

Reply:
As usual, I misread your comment. I'll make sure to add such a test.

Reply:
Well, this was interesting. Contrary to my expectations, the exception group in `__context__` on py3.11 was a _copy_ of the original, so `assert exc.value.__context__ is excgrp` failed.

Reply:
I'm not sure that it's essential to copy the exception group. I stopped just short of doing that, as I'm not sure it's worth the effort to change the context to a copy.